File: src/data/courses/dsa/essential-math/core-modular-ops/modular-operations/code/python/modular_operations.py
MOD=10**9+7
def add(a, b): return ((a%MOD)+(b%MOD))%MOD
def sub(a, b): return ((a%MOD)-(b%MOD)+MOD)%MOD
def mul(a, b): return ((a%MOD)*(b%MOD))%MOD
def pow_mod(a, b): r=1; a%=MOD; [(r:=r*a%MOD if b&1 else r) or (a:=a*a%MOD) or (b:=b>>1) for _ in iter(lambda:b,0)]; return r
print(f"(10+20)%MOD: {add(10,20)}"); print(f"2^10%MOD: {pow_mod(2,10)}")

File: src/data/courses/dsa/essential-math/core-modular-ops/modular-operations/code/java/ModularOperations.java
public class ModularOperations {
    long MOD=1000000007;
    long add(long a, long b) { return ((a%MOD)+(b%MOD))%MOD; }
    long sub(long a, long b) { return ((a%MOD)-(b%MOD)+MOD)%MOD; }
    long mul(long a, long b) { return ((a%MOD)*(b%MOD))%MOD; }
    long pow(long a, long b) { long r=1; a%=MOD; while(b>0) { if((b&1)==1) r=(r*a)%MOD; a=(a*a)%MOD; b>>=1; } return r; }
    public static void main(String[] args) { ModularOperations m=new ModularOperations(); System.out.println("(10+20)%MOD: "+m.add(10,20)); System.out.println("2^10%MOD: "+m.pow(2,10)); }
}